x/campaign/types/total_supply.go

```diff
@@ -21,3 +25,34 @@ func UpdateTotalSupply(coins, updatedCoins sdk.Coins) sdk.Coins {
 
 	return append(updatedCoins, notUpdated...).Sort()
 }
+
+// ValidateTotalSupply checks whether the total supply for each denom is within the provided total supply range
+func ValidateTotalSupply(coins sdk.Coins, supplyRange TotalSupplyRange) error {
+	// extra safety check, this should never happen
+	if supplyRange.MaxTotalSupply.LT(supplyRange.MinTotalSupply) {
+		return fmt.Errorf(
+			"provided total supply range is invalid, min > max: [%s, %s]",
+			supplyRange.MinTotalSupply.String(),
+			supplyRange.MaxTotalSupply.String(),
+		)
+	}
+
+	for _, coin := range coins {
+		totalSupply := coin.Amount
+		if totalSupply.LT(supplyRange.MinTotalSupply) {
+			return fmt.Errorf(
+				"total supply for %s cannot be less than minTotalSupply %s: %s",
```

Comment:
```suggestion
				"total supply for %s cannot be lower than minTotalSupply %s: %s",
```

Reply:
This is arbitrary, it's american vs british english imho (but I don't think we have chosen one or the other for style afaik), but fine I'll change it